internal/controller/postgrescluster/instance.go

```diff
@@ -1200,25 +1200,33 @@ func (r *Reconciler) reconcileInstance(
 			spec, instanceCertificates, instanceConfigMap, &instance.Spec.Template)
 	}
 
-	if err == nil &&
-		(feature.Enabled(ctx, feature.OpenTelemetryLogs) && !feature.Enabled(ctx, feature.OpenTelemetryMetrics)) {
+	if err == nil {
 
-		// TODO: Setting the includeLogrotate argument to false for now. This
-		// should be changed when we implement log rotation for postgres
-		collector.AddToPod(ctx, cluster.Spec.Instrumentation, cluster.Spec.ImagePullPolicy, instanceConfigMap, &instance.Spec.Template.Spec,
-			[]corev1.VolumeMount{postgres.DataVolumeMount()}, "", false)
-	}
+		// If OTel metrics are enabled, we'll need the MonitoringUserSecret
+		var password = ""
+		if feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+			monitoringUserSecret := &corev1.Secret{ObjectMeta: naming.MonitoringUserSecret(cluster)}
+			err = errors.WithStack(
+				r.Client.Get(ctx, client.ObjectKeyFromObject(monitoringUserSecret), monitoringUserSecret))
 
-	if err == nil &&
-		feature.Enabled(ctx, feature.OpenTelemetryMetrics) {
+			if err == nil {
+				password = string(monitoringUserSecret.Data["password"])
+			}
+		}
 
-		monitoringUserSecret := &corev1.Secret{ObjectMeta: naming.MonitoringUserSecret(cluster)}
-		err = errors.WithStack(
-			r.Client.Get(ctx, client.ObjectKeyFromObject(monitoringUserSecret), monitoringUserSecret))
+		// FIXME: is this bool really asking for a backup vol?
```

Comment:
Yeah, gotta figure this out tomorrow

Reply:
Looked into this -- it's not actually looking for a vol, just that backups are in the spec, so I'll use `pgbackrest.RepoHostVolumeDefined`